### PR TITLE
Add Scala version which can sort custom type.

### PR DESCRIPTION
**Pull request by KentarouSuzuki**
I implement stalin sort in Scala lang.
This implemention can sort custom type.

If you run it, result will be following.
```
List(1, 3, 4, 5, 10)
List(1, 3, 4, 5, 10)
List(1.11, 3.33, 4.44, 5.55, 10.1)
List(1.11, 3.33, 4.44, 5.55, 10.1)
List(Vegetable(cucumber,45), Vegetable(eggplant,80), Vegetable(cabbage,100), Vegetable(lettuce,120))
```

### CODE REVIEW

**Comment by gustavo-depaula** [comment]
great work! can you post the result of the list `1 3 3  5 4` and change the name of your file to CustomTypeStalinSort?

File: scala/CustomTypeStalinSort.scala [hunk 1]
```diff
@@ -66,4 +66,6 @@ object CustomStalinSort extends App {
   }
 
   println(vegetableList.stalinSort)
-}
+
+  println(Seq(1, 3, 3, 5, 4).stalinSort)  //if you run it, the result of is List(1, 3, 5)
```
Comment (gabrielcarneiro97) [inline scala/CustomTypeStalinSort.scala]: Hey mate, as is described at the README, we expect that equal numbers were preserved, could you update your code to this approach?

Comment (gabrielcarneiro97) [reply scala/CustomTypeStalinSort.scala]: The expected result is (1, 3, 3, 5)

Comment (KentarouSuzuki) [reply scala/CustomTypeStalinSort.scala]: Oh...thank you for your pointing out!!
I overlooked this condition.
I will update this code.

**Comment by KentarouSuzuki** [comment]
@gustavo-depaula @gabrielcarneiro97 
I have fixed code.
Please check it!

**Comment by gabrielcarneiro97** [comment]
All right! Thanks for you collaboration, feel free to help us with any other language that you're familiar with!

**Pull request merged**